tests/test_model_upload.py

```diff
@@ -1,8 +1,122 @@
+    def do_HEAD(self):  # noqa: N802
+        self.send_response(200)
+
+    def do_GET(self):  # noqa: N802
+        if self.path == f"/api/v1{GET_MODEL}":
+            self.send_response(200)
+            self.send_header("Content-type", "application/json")
+            self.end_headers()
+            self.wfile.write(bytes(json.dumps({"message": "Model exists!"}), "utf-8"))
+        elif self.path == f"/api/v1{GET_INSTANCE}":
+            self.send_response(200)
+            self.send_header("Content-type", "application/json")
+            self.end_headers()
+            self.wfile.write(bytes(json.dumps({"message": "Instance exists!"}), "utf-8"))
+        else:
+            self.send_response(404)
+            self.send_header("Content-type", "application/json")
+            self.end_headers()
+            response_data = {"message": "Some response data"}
+            self.wfile.write(bytes(json.dumps(response_data), "utf-8"))
+
+    def do_POST(self):  # noqa: N802
+        instance_or_version = self.path.split("/")[-1]
+        if self.path == f"/api/v1{CREATE_MODEL}":
+            self.send_response(200)
+            self.send_header("Content-type", "application/json")
+            self.end_headers()
+            self.wfile.write(json.dumps({"status": "success", "message": "Model created successfully"}).encode("utf-8"))
+        elif instance_or_version in ("instance", "version"):
+            self.send_response(200)
+            self.send_header("Content-type", "application/json")
+            self.end_headers()
+            response = {"status": "success", "message": "Model Instance created successfully"}
+            self.wfile.write(json.dumps(response).encode("utf-8"))
+        elif self.path == "/api/v1/blobs/upload":
+            self.send_response(200)
+            self.send_header("Content-type", "application/json")
+            self.end_headers()
+            self.wfile.write(
+                json.dumps(
+                    {
+                        "token": "dummy",
+                        "createUrl": "http://localhost:7778",
+                        "status": "success",
+                        "message": "Here is your token and Url",
+                    }
+                ).encode("utf-8")
+            )
+        else:
+            self.send_response(200)
+            self.send_header("Content-Type", "application/json")
+            self.end_headers()
+            error_message = json.dumps({"error": f"bad: {self.path}"})
+            self.wfile.write(bytes(error_message, "utf-8"))
+
+
+class GcsAPIHandler(BaseHTTPRequestHandler):
+    def do_HEAD(self):  # noqa: N802
+        self.send_response(200)
+
+    def do_PUT(self):  # noqa: N802
+        self.send_response(200)
+        self.send_header("Content-type", "application/json")
+        self.end_headers()
+        self.wfile.write(json.dumps({"status": "success", "message": "File uploaded"}).encode("utf-8"))
+
 
 class TestModelUpload(BaseTestCase):
```

Comment:
Can you add an extra test case for when you have more than 50 files? Ensure it throws an exception.